### ISSUE TITLE
can't set up environment

### ISSUE THREAD

**Issue by JuanTan628** [theislab/single-cell-tutorial#59]
Hi, 
I was trying to follow your instruction to set up the environment on win 10 using 
_conda env create -f sc_tutorial_environment.yml_
But an error pops up.
_Collecting package metadata (repodata.json): done
Solving environment: failed
ResolvePackageNotFound:
  - bioconductor-rhdf5lib_
Is it because of my platform? Any idea how to fix this? 
Thanks.

**Comment by LuckyMD** [comment]
Hi @JuanTan628,

Unfortunately, i have no experience with setting up these environments on windows, and I don't have a windows machine to test this. Would the docker image provided in the main repo be an alternative way to set up the environment instead?

**Comment by DuaaAlawad** [comment]
> Hi,
> I was trying to follow your instruction to set up the environment on win 10 using
> _conda env create -f sc_tutorial_environment.yml_
> But an error pops up.
> _Collecting package metadata (repodata.json): done
> Solving environment: failed
> ResolvePackageNotFound:
> 
> * bioconductor-rhdf5lib_
>   Is it because of my platform? Any idea how to fix this?
>   Thanks.

I have the same issue. Anyone can help me with this?

**Comment by LuckyMD** [comment]
You could use the docker image provided in the repo to run your analysis via that image?

**Comment by JakeLehle** [comment]
Hi,

I've had the same issues. I have a Linux machine at home and a windows laptop that I use in the lab so I've been running into this same issue when I'm at work. I looked into the sc_tutorial_environment.yml file and it just lists out that that bioconductor-rhdf5lib as a dependency.  I can open up R and install the rhdf5 package through bioconductor as I would normally. So here is what I'm thinking. 

1. Would it be possible to remove that dependency line from the .yml file and set up the environment without it? (I've been trying to do that and the wait time is super long >1hr. Seems like it is getting stuck.) **How long should it take to set up this environment in Anaconda normally?** 
2. If you can set up the environment without the but now you have to have that package installed couldn't you install Rstudio and then open that up and call in the package through that? If that would work it should have the package in that local environment I would think.

Let me know if that makes any sense I'm trying a few things out to fix this but I'm mainly getting stuck waiting for the >conda env create -f sc_tutorial_environment.yml command to go through. If this works, I'll make sure to post it back on here so people can try that as a fix in the future.

**Comment by JakeLehle** [comment]
Hi again,

Okay, I tried a few different things to fix this, and unfortunately, I gave up trying to run the code on a Windows OS. My workaround for this was to install the Windows subsystems for Linux (WSL) and run Debian Linux natively inside of windows and then manually install all of the packages required for the environment.

If you don't mind working solely in a terminal window then installing WSL on your Windows computer is a good option. However, if you want to additionally use something like spyder to run the python code you will have to set up a X11 graphic server and export out the DISPLAY to the localhost display port so you can run the code with the spyder GUI interface. There are many tutorials on how to set this up if you google "Xming to WSL" or "Xming to a Linux SSH".

**Comment by LuckyMD** [comment]
Hi @JakeLehle,

Sorry for being silent for so long here. I'm afraid I can't be of much help for setting up anything on Windows OS. It is in general not uncommon for the installation especially of R packages to take a while (an hour is very possible).

It is possible to install certain packages in `R` separately... this is what we're doing here as well for those that are not available in conda (`devtools`, `gam`, `scran`, etc).

Also, sorry that you had to brute for the installation via Linux in the end. Does the docker option not work (should be doable via VMware and docker for windows last time I heard... maybe better now)? We added the docker image to make the env as portable across machines as possible.

**Comment by JakeLehle** [comment]
Hi @LuckyMD 

No, you are fine. Thanks for getting back to me. So I saw you mention trying the docker option but I've never used that so I was holding off on looking into it. I just looked it up, and yeah that would have saved me a couple of hours of time haha.
But it was fun setting up a Linux machine again. In the future, it looks like using a docker image is the fastest way to port the environment over and I recommend anyone reading this to look into it as a possible option to fix the environment problem.
